### PR TITLE
Verify correct additional location for reported diagnostic

### LINKED ISSUE #46728: Bug in DiagnosticAnalysisContextHelpers.VerifyDiagnosticLocationsInCompilation
📝 This was a bug fix. Do we have new tests covering it?

_Originally posted by @sharwell in https://github.com/dotnet/roslyn/pull/46409_

### PR DESCRIPTION
**Pull request by mavasani**
Fixes #46728

### CODE REVIEW

File: src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContextHelpers.cs [hunk 1]
```diff
@@ -80,7 +80,7 @@ internal static void VerifyDiagnosticLocationsInCompilation(Diagnostic diagnosti
             {
                 foreach (var location in diagnostic.AdditionalLocations)
                 {
-                    VerifyDiagnosticLocationInCompilation(diagnostic.Id, diagnostic.Location, compilation);
+                    VerifyDiagnosticLocationInCompilation(diagnostic.Id, location, compilation);
```
Comment (mavasani) [inline src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContextHelpers.cs:83]: This is the only product change

**Comment by mavasani** [comment]
@dotnet/roslyn-compiler for reviews - trivial one line fix

**Review by cston — APPROVED** [review]

**Comment by mavasani** [comment]
@dotnet/roslyn-compiler Ping for second review - trivial one line change.

**Comment by mavasani** [comment]
@dotnet/roslyn-compiler Ping for second review - trivial one line change.

**Comment by mavasani** [comment]
@dotnet/roslyn-compiler Ping for second review

**Review by 333fred — APPROVED** [review]

**Comment by mavasani** [comment]
Thank you @333fred!

**Pull request merged**